front/backup.php

```diff
@@ -62,65 +62,50 @@
 
 
 /**
- * Genere un fichier backup.xml a partir de base dbhost connecte avec l'utilisateur dbuser
- * et le mot de passe dbpassword sur le serveur dbdefault
-**/
+ * Generate an XML backup file of the database
+ * @global DB $DB
+ * @global string $path Backup storage path
+ */
 function xmlbackup() {
-   global $CFG_GLPI, $DB;
-
-   //on parcoure la DB et on liste tous les noms des tables dans $table
-   //on incremente $query[] de "select * from $table"  pour chaque occurence de $table
-
-   $result = $DB->list_tables();
-   $i      = 0;
-   while ($line = $DB->fetch_row($result)) {
-      $table = $line[0];
-
-      $query[$i] = "SELECT *
-                    FROM `$table`";
-      $i++;
+   global $DB, $path;
+
+   // Will contain the select * from $table
+   $query = [];
+   // Foreach DB tables...
+   foreach ($DB->listTables() as $id => $row) {
```

Comment:
Do not call method in the loop, store in a variable before and use that on in the loop.